File: src/main/ngapp/src/app/score.ts
import {Player} from "./player";

export class Score {
  private readonly _player: Player;
  private _score: number = 0;

  constructor(player: Player) {
    this._player = player;
  }

  get player(): Player {
    return this._player;
  }

  set score(value: number) {
    this._score = value;
  }

  get score(): number {
    return this._score;
  }
}

File: src/main/java/game/actors/Score.java
package game.actors;

import requests.spark.websockets.objects.JSONWeb;
import requests.spark.websockets.objects.messages.mapping.WSData;
import requests.spark.websockets.objects.messages.mapping.WSDataReference;
import requests.spark.websockets.objects.messages.mapping.WSDataTypeScriptClass;

public class Score extends JSONWeb {
    @WSDataReference(WSData.SCORE_SCORE)
    private Integer score = 0;

    @WSDataReference(WSData.SCORE_PLAYER)
    @WSDataTypeScriptClass(Player.class)
    private Player player;

    public Score(Player player) {
        this.player = player;
    }

    public Player getPlayer() {
        return player;
    }

    public Integer getScore() {
        return score;
    }

    public Score givePoint() {
        score++;
        return this;
    }

    public void setScore(Integer score) {
        this.score = score;
    }

    public void setPlayer(Player player) {
        this.player = player;
    }
}
